src/authorized_keys_d/authorized_keys_d.go

```diff
@@ -259,11 +261,32 @@ func (akd *SSHAuthorizedKeysDir) Disable(name string) error {
 	return ak.Disable()
 }
 
-// Add adds the supplied key at name.
+// validateKeys validates keys via ssh.ParseAuthorizedKey.
+func validateKeys(keys []byte) error {
+	_, _, _, rest, err := ssh.ParseAuthorizedKey(keys)
+	if err == nil && len(rest) > 0 {
+		err = validateKeys(rest)
```

Comment:
if I'm reading this right I think this is going to fail with `"ssh: no key found"` if the input contains multiple trailing empty lines or a trailing comment line. This wouldn't be a problem if that error was defined statically in the module so you could add a `if err == NoKeysFound { err = nil }` after that recursive call but sadly `errors.New` gets called every time.

Since as-is this API is flat-out not usable as a validator due to this limitation which boggles my mind but it would be an easy fix to submit upstream.

Or I just read it wrong. Perhaps some unit tests to prove one way or the other? :)